sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py

```diff
@@ -97,7 +97,7 @@ def settings(self) -> PipelineJobSettings:
 
     @settings.setter
     def settings(self, value):
-        if value is not None and not isinstance(value, PipelineJobSettings):
+        if value and not isinstance(value, PipelineJobSettings):
```

Comment:
why this change resolves the issue?

Reply:
In this bug, I find that we would pass an empty dict here and empty dict is not None. Therefore I modify here.